### DISCUSSION TITLE
Misorientation in specimen coordinates after selecting a single grain.

### DISCUSSION THREAD

**Discussion by PennyWieser** [mtex-toolbox/mtex#2179] [Ask Anything] [answered]
I have an EBSD map that has multiple grains. 
I am trying to get the misorientation axis in crystal and specimin coordinates to compare to the grain boundary trace. 

I have first calculated grains 

`[grains,ebsd.grainId,ebsd.mis2mean]= calcGrains(ebsd,'angle',[subgrain_angle*degree,grain_div_angle*degree]);
`

I have then selected the grain (k) based on the GOS value, and calculated grains again (I'm guessing this isnt necessary but I was having more problems before I did this). 

`currentGrainIdsz = grains(k).id;  % IDs of pixels in the current grain`

` ebsd = ebsd(ebsd.grainId == currentGrainIdsz);  % Subset EBSD data for current grain`

 `[grains,ebsd.grainId,ebsd.mis2mean]= calcGrains(ebsd,'angle',[subgrain_angle*degree,grain_div_angle*degree]);`

Now, I want to get the misorientation axes of the inner boundaries in specimen and crystal coordinates. 

`gb_C=grains.innerBoundary('Fo', 'Fo')`

% Calculate the direction of the misorientation axis in crystal coordinates
`gB_axesC=gb_C.misorientation.axis;`

%Calculate the direction of the misorientation axis in specimin coordinates. 
`o=ebsd(gb_C.ebsdId).orientations;`
`ROT=axis(o(:,1),o(:,2));`

%Calculate grain boundary directions in specimin coordinates
`GBD=gb_C.direction;`

the o= line throws the following error. 
![image](https://github.com/user-attachments/assets/5f43b121-0e67-456f-96ca-b409c8f15508)

I gather that this is a fix
`o=ebsd('id', gb_C.ebsdId).orientations`

But I'm not really sure I follow at what point the indices are getting misaligned?

**Comment by zmichels** [comment]
Does it work if you change to include 'id'? 

Like:
ebsd('id', gb_C.ebsdId)

**Comment by kilir** [comment, marked as answer]
There are several ways top do that:
```
mtexdata forsterite

[grains,ebsd.grainId,ebsd.mis2mean]= calcGrains(ebsd,'angle',[10,1]*degree);

% id of interst
k = 17886

% inner boundary of grain
iB = grains('id',k).innerBoundary;


% angle
mAng = iB.misorientation.angle;
cond = mAng > 1.5*degree;

% axis in crystal coords
axC = iB.misorientation.axis;

% axis in specimen coords
o = ebsd('id',iB.ebsdId).orientations;
axS = axis(o(:,1),o(:,2))
axS.antipodal=1;

% plot
plot(grains('id',k).boundary)
hold on
ckc = HSVDirectionKey(axC.CS)
plot(iB(cond),ckc.direction2color(axC(cond)))
hold off

nextAxis
plot(axC(cond).project2FundamentalRegion,ckc.direction2color(axC(cond)),'fundamentalRegion')

nextAxis
plot(grains('id',k).boundary)
hold on
cks = HSVDirectionKey(specimenSymmetry('-1'))
plot(iB(cond),cks.direction2color(axS(cond)))
hold off

nextAxis
plot(axS(cond),cks.direction2color(axS(cond)),'fundamentalRegion','antipodal')
```
